review08/README.md

```diff
@@ -0,0 +1,241 @@
+2. `@Liveness` annotations -- Checks if the service is running.
+3. `@Readiness` annotations -- Checks if the service is connected to the database/Kafka,... without any additional code and is ready to serve requests.
+4. `@Startup` annotations -- Option for slow starting containers to delay an invocation of the liveness check.
+
+## Monitoring and Metrics
+
+Observability is a key feature of the service. It's important to monitor what is happening inside the application to be able to react to the problems and to improve the performance.
+
+### What do we want to measure?
+
+- **Uptime** -- How long the service is running.
+- **Rate of requests** -- How many requests are coming to the service.
+- **Latency** -- How long it takes to process the request.
+- **Error rate** -- How many requests are failing.
+- **Memory usage** -- How much memory is used.
+- **CPU usage** -- How much CPU is used.
+- **Network usage** -- How much data is sent and received.
+- **Custom metrics** -- Business-specific metrics.
+
+### Custom metrics
+
+Business specific metrics are important to understand the state of the application. For example, how many passengers are being onboarded, how much baggage is lost, etc.
+
+## Tools for metrics and monitoring
+
+![Monitoring architecture](./img/metrics_architecture.jpg)
+
+
+### Micrometer metrics
+
+Micrometer metrics provide a way to collect metrics and send them to the monitoring system. With the `micrometer-registry-prometheus` extension, we can send metrics to Prometheus, such as memory usage, CPU usage, latency, custom metrics, etc.
+
+Micrometer monitors the state of the application and exposes those metrics on the `/q/metrics` endpoint for Prometheus to scrape.
+
+### Prometheus
+
+Prometheus is a monitoring system that collects metrics from the services and stores them in a database. It provides a way to query and visualize the metrics.
+
+It pulls the metrics data from the `/q/metrics` endpoint of the services every few seconds and stores them in the database for further analysis. We can then write a query in PromQL that can be visualized in a visualization solution like Grafana.
+
+Prometheus can also send alerts if the metrics are not in the expected range.
+
+#### Custom metrics
+
+- **Gauge** -- A metric that represents a single numerical value that can arbitrarily go up and down.
+- **Counter** -- A metric that represents a single numerical value that only ever goes up. Cumulative in nature.
+- **Summary** -- A metric to represent the distribution of a set of values.
+- **Timer** -- A metric to measure the duration of a particular event.
+
+#### Example custom metrics
+
+```java
+public class PassengerResource {
+    
+    @Counted(value = "passenger_onboard", description = "How many passengers are onboarded")
+    public void onboardPassenger() {
+        // ... 
+    }
+    
+    // or
+   
+    @Timed(value = "passenger_onboard_duration", description = "How long it takes to onboard a passenger")
```

Comment:
just if you default to annotation, I would mention that only counter and timer exist, and not `@Gauge` or `@Summary`